src/providers/ad/ad_pac_common.c

```diff
@@ -21,6 +21,9 @@
 */
 
 
+#include <sys/types.h>
+#include <sys/time.h>
```

Comment:
What for?
Also note you already added time.h to ad_pac.h that is included here.